tsdb/head.go

```diff
@@ -987,6 +1004,72 @@ func (h *Head) truncateMemory(mint int64) (err error) {
 	return nil
 }
 
+func (h *Head) WaitForPendingReadersInTimeRange(mint, maxt int64) {
+	// TODO: should we have some limit on how long we wait in total? If so, how long?
+	overlaps := func() bool {
+		o := false
+		h.iso.TraverseOpenReads(func(s *isolationState) bool {
+			if s.mint <= maxt && mint <= s.maxt {
+				// Overlaps with the truncation range.
+				o = true
+				return false
+			}
+			return true
+		})
+		return o
+	}
+	for overlaps() {
+		time.Sleep(500 * time.Millisecond)
+	}
+}
+
+// IsQuerierCollidingWithTruncation tells if the current querier should be closed and if
+// have to get a new querier. If should get a new querier, it also tells
+// what is the new mint of the Head to be considered.
+// This function helps in preventing race with truncation of in-memory data.
```

Comment:
Suggestion to iron out a few language quirks:

"""
IsQuerierCollidingWithTruncation returns if the current querier needs to be closed and if a new querier has to be created. In the latter case, the method also returns the new mint to be used for creating the new range head and the new querier. This methods helps preventing races with the truncation of in-memory data.
"""

I also recommend to leave an empty comment line after this (`// `) because Godoc will otherwise render the `NOTE` below as part of the same paragraph, reducing its visibility.